test/communication_test.go

```diff
@@ -25,15 +25,16 @@ var testGrpcClientConn *grpc.ClientConn
 // connection is then re-used in all the tests to reduce overhead.
 // Once all the tests run, we close the connection to the server and stop the server.
 func TestMain(m *testing.M) {
-	// Set debug log level for easier troubleshooting in case something goes wrong
-	client.SetLogLevel("debug")
-	server.SetLogLevel("debug")
-
-	server, err := server.NewProtocolServer("testdata/server.pem", "testdata/server.key")
+	server, err := server.NewProtocolServer("testdata/server.pem", "testdata/server.key", "")
 	if err != nil {
 		fmt.Println(err)
 		os.Exit(1)
 	}
+
+	// Set debug log level for easier troubleshooting in case something goes wrong
+	client.SetLogLevel("notice")
+	server.SetLogLevel("notice")
```

Comment:
Here in the test you could then have yet another logger implemented that would just remember all the messages obtained so far (without printing them) and only print them if an error occurs 😄